src/mcp_atlassian/server.py

```diff
@@ -1882,6 +1913,46 @@ def format_comment(comment: Any) -> dict[str, Any]:
                 )
             ]
 
+        elif name == "jira_batch_create_issues" and ctx and ctx.jira:
+            if not ctx or not ctx.jira:
+                raise ValueError("Jira is not configured.")
+
+            # Write operation - check read-only mode
+            if read_only:
+                return [
+                    TextContent(
+                        "Operation 'jira_batch_create_issues' is not available in read-only mode."
+                    )
+                ]
+
+            # Extract required arguments
+            issues = arguments.get("issues")
+            validate_only = arguments.get("validate_only", False)
+
+            # Parse issues from JSON string to list of dictionaries
+            if issues and isinstance(issues, str):
+                try:
+                    issues = json.loads(issues)
+                except json.JSONDecodeError:
+                    raise ValueError("Invalid JSON in issues")
+
+            # Create issues in batch
+            created_issues = []
+            for issue in issues:
+                created_issues.append(ctx.jira.create_issue(**issue))
```

Comment:
This remains the critical flaw. The code here iterates through the `issues` list and calls the *single* issue creation method (`ctx.jira.create_issue`) for each item. This completely bypasses the new `batch_create_issues` mixin method and the underlying Jira bulk creation API (`POST /rest/api/X/issue/bulk`) that the mixin method correctly uses via `self.jira.create_issues`.

To fix this and achieve the intended batch creation, you must call the `batch_create_issues` method directly.

```suggestion
            created_issues = ctx.jira.batch_create_issues(
                issues=issues, validate_only=validate_only
            )
```

Reply:
Ah sorry this was an oversight, you are absolutely correct. Fixed!